README.md

```diff
@@ -1,168 +1,222 @@
+```
 
-Run the app with `ng s`. The demo app runs on <http://0.0.0.0:4200> and we use it for documentation on [Knora-ui Github page](https://dasch-swiss.github.io/knora-ui).
+will create a component-folder called `test` inside of `projects/dsp-ui/src/lib/core/` with four files:
 
-There's a test environment for the modules on <https://github.com/dhlab-basel/knora-ui-playground> with yalc.
+- `test.component.scss`
+- `test.component.html` 
+- `test.component.spec.ts`
+- `test.component.ts`
 
-* * *
+The main component file should look as follow:
 
-#### YALC
+```typescript
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'kui-test',
+  templateUrl: './test.component.html',
+  styleUrls: ['./test.component.scss']
+})
+export class TestComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+}
+```
 
-> Better workflow than `npm` \| `yarn` link for package authors.
+Before testing the new component inside of the demo app, you have to rebuild after each change: 
 
-Yalc publishes the packages to a local store (not the npm website).
+```bash
+npm run build-lib
+```
+
+### Run the DSP-UI-LIB Demo Application
+
+Run the app with the command line: `ng s`.
+
+The demo app runs on <http://0.0.0.0:4200> and we use it for documentation on [DSP-UI-LIB Github page](https://dasch-swiss.github.io/knora-ui).
+
+There's an additional test environment for the modules on <https://github.com/dhlab-basel/knora-ui-playground> with yalc.
+
+#### YALC
+
+In some cases we have to work with unpublished npm packages like our own [@knora/api](https://github.com/dasch-swiss/knora-api-js-lib), a JavaScript library that allows a developer to implement the DSP API without knowing technical details about it.
+To publish and add local packages we use [yalc](https://www.npmjs.com/package/yalc). Yalc publishes the packages to a local store (not the npm website).
 From there, the packages can be added to your depending project.
 
-#### Install [yalc](https://github.com/whitecolor/yalc):
+##### Installation
 
 ```bash
-$ npm i yalc -g
+npm i yalc -g
 ```
 
-#### Usage
+##### Usage
 
 Publish library to local store:
 
 ```bash
-$ npm run build-app
-$ npm run yalc-publish
+npm run build-app
+npm run yalc-publish
 ```
 
 Use them in your application:
 
 ```bash
-$ yalc add @knora/ui
+yalc add @dasch-swiss/dsp-ui
+npm install
+```
+
+To remove from the project and restore `package.json`, run:
+
+```bash
+yalc remove --all
 ```
 
-To remove from the project and restore `package.json` run:
+#### Conflicts
+
+In case of an issue with package-lock.json (e.g. merge conflict) where you have to reset/delete package-lock file, you must consider the following commands in the correct order:
 
 ```bash
-$ yalc remove --all
+yalc remove --all
+rm -rf node_modules
+rm package-lock.json
+npm install
+yalc add @knora/api
+npm install
 ```
 
-## Running the application in productive mode
+### Run the application in productive mode
```

Comment:
I think this should come right after the section Run the DSP-UI-LIB Demo Application